SuperTerrain+/CoreInterface/SuperTerrain+/Utility/STPDatabaseErrorHandler.hpp

```diff
@@ -0,0 +1,21 @@
+#include <SuperTerrain+/Utility/STPGenericErrorHandlerBlueprint.hpp>
+
+#ifdef SQLITE3_H
+#define STP_HAS_SQLITE3
+#endif
+
+//#include <sqlite3.h>
+#ifdef STP_HAS_SQLITE3
+#ifndef _STP_DATABASE_ERROR_HANDLER_HPP_
```

Comment:
It seems like a very funky include guard, but it prevent one case during our testing:

Include error handler without including a *sqlite* header, which emits no definition for error handler. Then include a *sqlite* followed by another include to this header. For traditional guard stays at the first line of the file, this will do nothing, no definition will be emitted either; in our implementation, it is prevented because guard is defined only when definition is available.